src/Plugins/PluginDiscovery.php

```diff
@@ -44,7 +43,8 @@ public function discover()
             foreach ($di as $dir) {
                 if ($dir->isDir() && !$dir->isDot() && $dir->isReadable()) {
                     try {
-                        $out[] = $this->getContainer()->get(PluginInfo::class, [$dir->getPathname()]);
```

Comment:
Calling `$container->get(...)` with call-time parameters is not defined in the [PSR-11 ContainerInterface](https://github.com/php-fig/container/blob/master/src/ContainerInterface.php#L23). As explained by [container-interop](https://github.com/container-interop/container-interop/blob/master/docs/ContainerInterface-meta.md), some container implementations (e.g. League/Container) have an additional parameter to the `get()` method, but its use is not consistent between containers.

Robo also uses this pattern, but it is not advisable to do so, and should be avoided.